### ISSUE TITLE
building issue on ubuntu lucid

### ISSUE THREAD

**Issue by GoogleCodeExporter** [tinselcity/skipfish#4]
```
on ubuntu lucid beta1, after installing libidn11-dev, it is still not 
possible to compile skipfish 1.03b.

we need to install libssl-dev package to make it compile, but the readme 
does not contain information about this.

What steps will reproduce the problem?
1. try make
2. you'll get openssl/ssl.h: No such file or directory error 
3. install libssl-dev, and now it compiles.


```

Original issue reported on code.google.com by `yig...@gmail.com` on 20 Mar 2010 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
Well, it's highly unusual for a development environment not to have openssl 
headers
installed; I can note this in the README, but...
```

Original comment by `lcam...@google.com` on 20 Mar 2010 at 2:58

**Comment by GoogleCodeExporter** [comment]
```
You also need zlib, etc; it's probably not worth covering, I think.
```

Original comment by `lcam...@gmail.com` on 20 Mar 2010 at 3:37
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
Try to install headers : 'apt-get install build-essential' before the make.
```

Original comment by `js.opdeb...@gmail.com` on 21 Mar 2010 at 10:08

**Comment by GoogleCodeExporter** [comment]
```
Oki Got it ... 'apt-get install libssl-dev' then it rocks.
```

Original comment by `js.opdeb...@gmail.com` on 21 Mar 2010 at 10:29

**Comment by GoogleCodeExporter** [comment]
```
Thanks to yigiit! I have successfully do "make" command. I am using Ubuntu 
9.10. 
```

Original comment by `zakiakh...@gmail.com` on 22 Mar 2010 at 7:24

**Comment by GoogleCodeExporter** [comment]
```
I have libssl-dev, zlib and build-essential but it wont compile under Ubuntu 
9.10. I
got this message:

$ make
cc skipfish.c -o skipfish -Wall -funsigned-char -g -ggdb -D_FORTIFY_SOURCE=0
-I/usr/local/include/ -I/opt/local/include/ -O3 -Wno-format http_client.c 
database.c
crawler.c analysis.c report.c -lcrypto -lssl -lidn -lz -L/usr/local/lib/ 
-L/opt/local/lib
http_client.c:39:18: error: idna.h: No such file or directory
http_client.c: In function ‘parse_url’:
http_client.c:276: warning: implicit declaration of function 
‘idna_to_ascii_8z’
http_client.c:276: error: ‘IDNA_SUCCESS’ undeclared (first use in this 
function)
http_client.c:276: error: (Each undeclared identifier is reported only once
http_client.c:276: error: for each function it appears in.)
make: *** [skipfish] Error 1

Any ideas?

```

Original comment by `garciaru...@gmail.com` on 22 Mar 2010 at 3:25

**Comment by GoogleCodeExporter** [comment]
```
how about libidn11-dev?
```

Original comment by `yig...@gmail.com` on 22 Mar 2010 at 3:29

**Comment by GoogleCodeExporter** [comment]
```
i do
apt-get install libssl-dev build-essential zlibc zlib-bin libidn11-dev libidn11
then make
and compile ok in Ubuntu 9.10 =)

By ToCaDo157
```

Original comment by `tocado....@gmail.com` on 22 Mar 2010 at 6:04

**Comment by GoogleCodeExporter** [comment]
```
tocado.157 thnx for to tip.
```

Original comment by `jsucupir...@gmail.com` on 22 Mar 2010 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
this should be included in the README file
```

Original comment by `dic...@gmail.com` on 4 Jun 2010 at 1:56

**Comment by GoogleCodeExporter** [comment]
```
comment # 8 works for me. (ubuntu 11.04). thanks !
```

Original comment by `christia...@gmail.com` on 26 May 2011 at 12:00

**Comment by GoogleCodeExporter** [comment]
```
#8 did the trick. I tried Synaptic but it didn't get the latest version. 
apt-get install got the latest versions and it worked. Thx! 
```

Original comment by `leung.be...@gmail.com` on 17 Jun 2011 at 11:20

**Comment by GoogleCodeExporter** [comment]
```
#8 helped me too. I didn't have libidn11-dev zlib-bin zlibc installed.
```

Original comment by `mozillal...@gmail.com` on 20 Nov 2011 at 9:02
